src/canvas/SelectableCanvas.ts

```diff
@@ -772,18 +774,19 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
     globalPointer: Point
   ): boolean {
     if (obj && obj.visible && obj.evented && obj.containsPoint(pointer)) {
-      if (
-        // if `obj` is selected we want it to be easy to interact with it
-        this.getActiveObject() !== obj &&
-        (this.perPixelTargetFind || obj.perPixelTargetFind)
-      ) {
-        if (!this.isTargetTransparent(obj, globalPointer.x, globalPointer.y)) {
-          return true;
-        }
-      } else {
-        return true;
-      }
+      const shouldPerformPixelFind =
+        !(obj as unknown as IText).isEditing &&
+        (this.perPixelTargetFind === true ||
+          obj.perPixelTargetFind === true ||
+          ((this.perPixelTargetFind === 'not-selected' ||
+            obj.perPixelTargetFind === 'not-selected') &&
```

Comment:
I am not sure about combinations between `canvas.perPixelTargetFind` and `obj.perPixelTargetFind`
Shouldn't the object take presedence?
I will update tests once we decide

Reply:
We should collect the current rules around the code.
But there shouldn't be such precedence is just confusing.
There must a be a documented way to handle that for all objects or just some.
If you can set it on a single object or an entire subclass through default values is good enough imho